React/events.md

```diff
@@ -8,84 +8,112 @@
 React supplies support on many different events. We'll focus on a few of the common events, keyboard and click, but just know that you should reference the [full list](https://reactjs.org/docs/events.html#supported-events) if you want more information.
 
 ### Context
-In this lecture, we are going to expand upon the `NameDisplay` component that we created in the [state lecture](state.md). This was a great component that we created that was storing a name and a boolean in the state. We used the boolean in the render function to determine what to display.
+
+In this lecture, we are going to expand upon the `Student` component that we created in the [state lecture](state.md). This was a great component that we created that was storing a student's present status in the component. We used the boolean in the function to determine what to display.
 
 ### Adding More Meaning
+
 #### Toggle our Boolean
-The boolean value that we're storing in the state seems a little bit useless right now. We set it to `true` within the constructor and we don't ever change it! We will set up a button and handle the `onClick` event to affect the state within this component.
 
-Let's add a button that we will use to toggle the display of the name variable. We'll first add the button element within the `return` of our `render` function. Then we'll add the `onClick` attribute whose value will refer to an event handler function that we will create next.
+The boolean value that we're storing in the state seems a little bit useless right now. We set it to `true` as an initial value and we don't ever change it! We have a button and it doesn't do anything!  Next we will add code to change the state of `present` when the button is clicked on.
+
+### onClick Event
+
+Now we'll add the `onClick` attribute to the button whose value will refer to an event handler function that we will create next.
 
 ```javascript
-// NameDisplay.js
-  render() {
-    return (
-      ...
-      <button onClick={ this.onButtonClick }>Toggle Display</button>
-    );
-  }
+// src/components/Student.js
+// ...
+
+// Function to toggle present
+const onButtonClick = () => onButtonClick(!present);
+
+  // Component functions always return JSX
+  return (
```

Comment:
I think it'd be helpful if we included the component definition for this code snippet so the context of the return is clear.
It took me a while to understand where this code was meant to fit in. (Even though in retrospect I do see you have a comment at the top to clarify that.

Reply:
done